test/node/conversation-labels.js

```diff
@@ -8,45 +8,39 @@ Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
 
 let client;
 let addedLabelsHT = {};
-describe('Authentication', () => {
+describe('Labels', () => {
 
-    it('should login using Client Credentials', async () => {
+    before(async () => {
         client = new Circuit.Client(config.bot1);
         const user = await client.logon();
-        assert(!!user);
     });
 
-    it('should be authenticated', async () => {
-        await client.isAuthenticated();
-    });
-
-
     it('should add two labels', async () => {
-        const labelValue1 = Date.now().toString() + 'a';
-        const labelValue2 = Date.now().toString() + 'b';
+        const labelValue1 = `${Date.now()}a`;
+        const labelValue2 = `${Date.now()}b`;
         const addedLabels = await client.addLabels([labelValue1, labelValue2]);
+        await helper.expectEvents(client, ['labelsAdded']);
```

Comment:
You should add the client.addLabels promise and  the helper.expectEvents promise into a Promise.all since you cannot guarantee that the addLabels returns before the event is received. See https://github.com/circuit/circuit-sdk/blob/master/test/browser/directcall.js#L63 for an example

Reply:
Might be better to also check for the label in the event. Use a predicate for that. see https://github.com/circuit/circuit-sdk/blob/master/test/browser/directcall.js#L36